packages/core/src/processed-data-manager.ts

```diff
@@ -0,0 +1,134 @@
+import get from 'lodash/get';
+import { ApplicationError } from 'n8n-workflow';
+import type {
+	IProcessedDataManager,
+	ICheckProcessedOptions,
+	ICheckProcessedOutput,
+	ICheckProcessedOutputItems,
+	IDataObject,
+	ProcessedDataContext,
+	ProcessedDataItemTypes,
+	ICheckProcessedContextData,
+} from 'n8n-workflow';
+
+export class ProcessedDataManager {
+	private static instance: ProcessedDataManager;
+
+	private manager: IProcessedDataManager;
+
+	private constructor(manager: IProcessedDataManager) {
+		this.manager = manager;
+	}
+
+	static async init(manager: IProcessedDataManager): Promise<void> {
+		if (ProcessedDataManager.instance) {
+			throw new ApplicationError('Processed Data Manager is already initialized');
+		}
+
+		ProcessedDataManager.instance = new ProcessedDataManager(manager);
+	}
+
+	static getInstance(): ProcessedDataManager {
+		if (!ProcessedDataManager.instance) {
+			throw new ApplicationError('Processed Data Manager is not initialized');
+		}
+		return ProcessedDataManager.instance;
+	}
+
+	static getManager(): string[] {
+		return Object.keys(ProcessedDataManager.instance.manager);
+	}
```

Comment:
This is an odd looking method. Is it used?

Reply:
removed